src/Build/BuildCheck/Infrastructure/TracingReporter.cs

```diff
@@ -6,13 +6,20 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Microsoft.Build.BuildCheck.Utilities;
+using Microsoft.Build.Experimental.BuildCheck;
 
 namespace Microsoft.Build.BuildCheck.Infrastructure;
 
 internal class TracingReporter
 {
     internal Dictionary<string, TimeSpan> TracingStats { get; } = new();
 
+    // Infrastructure time keepers, examples for now
```

Comment:
Can you help me understand how the stat that feels most important for a user, "how much time overhead did running buildchecks add?" will fit in here?

Reply:
Right now this is just a foundation for the instrumentation module. It does not contain all necessary measures that we will want to keep in the future, but it is just building the infrastructure behind it.